app/src/project/components/viewProject/viewProject.view.js

```diff
@@ -33,26 +51,50 @@ export class ViewProjectComponent extends Component {
-                        <img className='logo_view' src={Inovia} />
-                        <h2 className='project_title'>{title}</h2>
-                        <p className='author'>By {author}</p>
-                    </div>
-                    <div className='progress'>
-                        <CircularProgressbar percentage={Math.round((acquired / estimate) * 100)} />
-                        <p className='time_required'>Time required : {acquired}/{estimate}h</p>
-                        <Link to={`/give/${id}`}><RaisedButton backgroundColor='#4CAF50' label={'Give Time'}/></Link>
-                        <br/>
+                <form onSubmit={handleSubmit}>
+                    <img style={{ float: 'right' }} onClick={() => this.editOrSave()} src={this.state.editing ? SaveIcon : EditIcon} />
+                    <div className='header_view'>
+                        <div className='basic_infos'>
+                            <img className='logo_view' src={Inovia} />
+                            {this.state.editing ? <Field
+                                id="title" name="title" type="text"
+                                component={TextField}
+                                label={'Current title : ' + title}
+                            /> : <h2 className='project_title'>{title}</h2>}
+                            <p className='author'>By {author}</p>
+                        </div>
+                        <div className='progress'>
+                            <CircularProgressbar percentage={Math.round((acquired / estimate) * 100)} />
+                                {this.state.editing ? <Field
+                                    id="estimate" name="estimate" type="number"
+                                    component={TextField}
+                                    label={'Current estimate : ' + estimate}
+                                /> : <p className='time_required'>Time required : {acquired}/{estimate}h</p>}
+                            <Link to={`/give/${id}`}><RaisedButton backgroundColor='#4CAF50' label={'Give Time'}/></Link>
+                            <br/>
+                        </div>
                     </div>
-                </div>
-                <div className='description'>
-                    <div className='title_arrow'>
-                        <h2 className='title_desc'>Description</h2>
-                        <img className='arrow_desc' onClick={() => this.showDesc()} src={this.state.showDesc ? UpArrow : DownArrow} />
+                    <div className='description'>
+                        <div className='title_arrow'>
+                            <h2 className='title_desc'>Description</h2>
+                            <img className='arrow_desc' onClick={() => this.showDesc()} src={this.state.showDesc ? UpArrow : DownArrow} />
+                        </div>
+                        {this.state.showDesc && !this.state.editing && (<p dangerouslySetInnerHTML={this.createMarkup(description)}></p>)}
```

Comment:
Too dangerous, use a simpler markup like bbcode or slack-like or anything else.
We cannot risk having XSS injections here.